src/NServiceBus.TransactionalSession/BatchedMessageSession.cs

```diff
@@ -16,37 +16,74 @@ class BatchedMessageSession : IBatchedMessageSession
             this.dispatcher = dispatcher;
         }
 
-        public virtual async Task Commit(CancellationToken cancellationToken = default) =>
-            await dispatcher.Dispatch(new TransportOperations(pendingOperations.Operations), new TransportTransaction(), cancellationToken).ConfigureAwait(false);
+        public virtual async Task Commit(CancellationToken cancellationToken = default)
+        {
+            ThrowIfDisposed();
+            ThrowIfCommitted();
+
+            await dispatcher.Dispatch(new TransportOperations(pendingOperations.Operations), new TransportTransaction(),
+                cancellationToken).ConfigureAwait(false);
+
+            committed = true;
+        }
 
         public virtual async Task Send(object message, SendOptions sendOptions, CancellationToken cancellationToken = default)
         {
+            ThrowIfDisposed();
```

Comment:
Both validation methods are always called at the same time. Could we switch to a single one e.g. `ThrowIfNotValidForUse`?